src/garage/torch/modules/discrete_cnn_module.py

```diff
@@ -31,6 +31,8 @@ class DiscreteCNNModule(nn.Module):
         hidden_sizes (list[int]): Output dimension of dense layer(s) for
             the MLP for mean. For example, (32, 32) means the MLP consists
             of two hidden layers, each with 32 hidden units.
+        dueling (bool): Whether to use a dueling architecture for the
```

Comment:
can you make this a separate module/QFunction class instead?

Reply:
I can. It seems a bit overkill to create a separate module though since there will be lots of overlapping code.

Reply:
the parts of this module not hidden behind "if dueling" are basically all boilerplate. It's an antipattern to implement 2 classes in 1 and then switch between their implementations using a bool. 

In Python, we don't do `my_list = Container(1,2,3, list=True)` and `my_tuple = Container(1,2,3, list=False)`, we do `my_list = list(1,2,3)` and `my_tuple = tuple(1,2,3)`.

It's also more honest to the user about what dueling DQN is, which is basically an alternative QF architecture. If you want to do dueling DQN, you construct a dueling Q-function and pass it to the regular-old DQN algorithm.

@krzentner WDYT?

Reply:
Fair enough, I'll get that done.